### ISSUE TITLE
Vector of Images and IMages to Matrices

### ISSUE THREAD

**Issue by Tokazama** [JuliaImages/Images.jl#754]
I'm specifically looking for a clean way of turning a bunch of images into a matrix that shares memory with those same images so that I can take advantage of functions in Images.jl but also perform manipulations on the matrix without having to reshape it repeatedly. I wanted to see if this already existed or if someone had a better idea before I make an attempt.

I figured the workflow would be something like...
```
using Images
img1 = ImageMeta(rand(3,3,3))
img2 = ImageMeta(rand(3,3,3))

imgVector = someFunction1(img1, img2)
imgMatrix = someFunction2(imgVector)
```
Where `someFunction1` produces
```
struct ImageVector
    data::Vector{AbstractArray}  # would contain a 2x3x3x3 array in the example above
    properties::Dict{String,Any}  # properties shared among ImageMeta types put in function
end
```
and `someFunction2` produces a matrix that shares memory with imgVector but each row represents a single image.

**Comment by rdeits** [comment]
Perhaps you could just `channelview` on each image to get a 3-dimensional array view of each image, and then use something like https://github.com/ahwillia/CatViews.jl to create a concatenated 4-dimensional array view (which would share its data with each of the original images).

**Comment by Tokazama** [comment]
I tried out `CatViews` and it was pretty interesting. I also stumbled upon `StackedView` and found this to work pretty well. 
```
function img2mat(x...)
    reshape(StackedView(x...),(length(x), prod(size(x[1]))))
end
```
I haven't tested the output on anything more than some stuff from `LinearAlgebra`.

Would it make sense to have `StackedViews` documented in the "Conversion vs. views" portion of the documentation? I only found it because it's listed under "See also:" for `colorview`.

Thanks for the help!
